src/lexer.l

```diff
@@ -20,7 +20,10 @@ struct lexer_param;
 %s IN_BRACKET
 %s IN_BRACE
 %s IN_QQINTERP
+
 %x IN_QQSTRING
+%x C_COMMENT
```

Comment:
Let's leave the lexer changes out for now.  Adding new comment forms in the jq language should be separate from comment stripping in JSON parsing.